### PR TITLE
Feature/markdown editor codeblock syntax highlighting

### PR DESCRIPTION
**Pull request by krusche**
Add a new markdown command for code blocks and showdown-highlight to properly highlight code in the markdown editor

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Steps for Testing
1. Add code into a problem statement using the new code block command, e.g.
2. Check in the preview and on the exercise details page that the code is displayed properly and keywords are hightlighted
3. Try out code for different programming languages, e.g. Java, Python, ...

image

image

### CODE REVIEW

**Review by sebastiantpeter — APPROVED** [review]
Tested with Java, C, Haskell and Python on ts1. Works well.

**Review by airelawaleria — DISMISSED** [review]

**Review by marvinchimeka — APPROVED** [review]
tested on ts2. Works as intended

**Review by MaisiKoleni — DISMISSED** [review]
Great addition, always wanted to have that. Thanks!

Code looks good to me as far as I can tell, and the feature seems to work fine:
- **Problem Statement**
    image
- **Quiz Questions**
    image
- Does not work for Q&A, but as far as I know this is disabled for security reasons now. Code in Q&A is not that important either, just might be nice in the future.

Also tried some other languages, and those work fine as well. In terms of the highlighting, if also works for code snippets (see screenshots) which is good. I was a bit surprised that the numbers were not highlighted in the first example code and thought it was broken. But that is at most a minor issue with the highlighting framework itself.

**Comment by MaisiKoleni** [comment]
Likely not directly related but strange:
- **In the Couse Exercise Details**
   image
- **In the Q&A Instructor Overview**
   image

**Review by JohannesWeiss — APPROVED** [review]
Tested on ts2 on worked for all tested languages. Some rather specific parts (special hardware functions in C in that case) were not highlighted, but I don't think that is a real issue.

**Review by EneaGore — DISMISSED** [review]
looks good on ts2

**Review by simonzilker — APPROVED** [review]
Nice feature! Looks good on ts2!

**Review by valeriebucher — DISMISSED** [review]
Looks good.

File: src/main/webapp/app/shared/markdown-editor/markdown-editor.component.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
                         <label for="file-upload" class="markdown-editor__file-label d-inline">
                             <div class="row mx-0 align-items-baseline">
                                 <span class="col upload-subtitle">{{ 'artemisApp.markdownEditor.fileUpload' | translate }}</span>
-                                <a class="col-auto" href="https://guides.github.com/features/mastering-markdown/">
+                                <a class="col-auto" href="http://demo.showdownjs.com">
```
Comment (jpbernius) [inline src/main/webapp/app/shared/markdown-editor/markdown-editor.component.html:26]: I think this help link should point to a list of markdown formats. Demoing markdown is also possible in the preview on Artemis itself.

**Review by jpbernius — CHANGES_REQUESTED** [review]
Looks good, I have one remark regarding the updated help link:

**Review by tiffca — DISMISSED** [review]
Tested successfully

**Review by DenizShabani — DISMISSED** [review]
Looks good on TS1!

**Review by JannisBittlmayer — DISMISSED** [review]
Tested with Java and C++, works as intended

**Review by FlorianStang — DISMISSED** [review]
Tested on TS1, everything worked as intended.

File: src/main/webapp/app/shared/markdown-editor/markdown-editor.component.html (hunk 1 above)
Comment (krusche) [reply src/main/webapp/app/shared/markdown-editor/markdown-editor.component.html:26]: But this is the official standard that we support. I specifically checked it. the previous link did not show the exact version but a more generic one. And the demo page includes many examples which are not available as commands in our case. Finding them and trying those out might be helpful for users.

One issue I see is that the link is not very accessible. I bet no one found it until now

**Review by philgabr — APPROVED** [review]
Works as intended. Tested with Java, Python, C & Kotlin keywords.

**Pull request merged**